clang/test/ParserSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -Wno-unused %s
+
+namespace NS{};
+
+void f(int var) {
+  // expected-error@+1{{expected '(' after '__builtin_unique_stable_name'}}
+  __builtin_unique_stable_name int;
+  // expected-error@+1{{expected '(' after '__builtin_unique_stable_name'}}
+  __builtin_unique_stable_name {int};
+
+  __builtin_unique_stable_name(var);
+  // expected-error@+1{{use of undeclared identifier 'bad_var'}}
+  __builtin_unique_stable_name(bad_var);
+  // expected-error@+1{{use of undeclared identifier 'bad'}}
+  __builtin_unique_stable_name(bad::type);
+  // expected-error@+1{{no member named 'still_bad' in namespace 'NS'}}
+  __builtin_unique_stable_name(NS::still_bad);
```

Comment:
These tests are more Sema tests than Parser tests. It's good to keep one of the tests that checks `foo::bar` to show that we accept more than a simple identifier in terms of parsing behavior, but that should be a test that passes rather than one that fails because of lookup.

Reply:
Ok. Added a passing case and removed these from the parser test. Will move them to SemaSYCL.